### PR TITLE
550 radio

### LINKED ISSUE #550: Make component library not assign ids by default to radio buttons (or anywhere else) bc ids with spaces aren't valid and they break accessibility
Giving this 2 points because figuring out the relationship between form things in the component library is a bit complex-- but basically we just need to pass the parent ID of a RadioButtonGroup to its individual radio buttons as props (and modify it for each one so they're all unique) so they can associate the label and input correctly

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #550 

This is a refactor of most of our forms, so I'm gonna push this one to QA and run e2e testing against it (yay) before merging it!  Any manual review that anyone is inspired to do would also be nice.

### CODE REVIEW

File: client/src/components/Forms/ChildInfo/Fields/DisabilityServices.tsx [hunk 1]
```diff
@@ -1,50 +1,24 @@
...
-} from '@ctoec/component-library';
-import { Child } from '../../../../shared/models';
-import { getValidationStatusForField } from '../../../../utils/getValidationStatus';
 import { UNKNOWN } from '../../../../shared/constants';
+import { Child } from '../../../../shared/models';
+import { TripleBooleanRadio } from '../../../TripleBooleanRadio';
 
 export const DisabilityServices: React.FC = () => (
-  <FormField<Child, RadioButtonGroupProps, boolean | null>
+  <TripleBooleanRadio<Child>
```
Comment (jlhogan) [inline client/src/components/Forms/ChildInfo/Fields/DisabilityServices.tsx:7]: Is tripleBooleanRadio a form field? I guess i missed the PR for that, but if it is a form field i think that should be expressed in the name? Does it use `field` prop to find/update data in the object? Why opt for magic string instead of type safe object driller?

Comment (mmazanec22) [reply client/src/components/Forms/ChildInfo/Fields/DisabilityServices.tsx:7]: TripleBooleanRadio is a component added to this project in this PR.  It's not a form field bc radio group is no longer a form field.  There isn't any more magic string-ness going on here than there was before, it's just condensed into one place for any radio button group where we're using the same logic

File: client/src/components/TripleBooleanRadio.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+export type TripleBooleanRadio<T> = {
+  field: keyof T;
+  id: string;
+  legend: string;
+  trueOption: TripleBooleanRadioOption;
+  falseOption: TripleBooleanRadioOption;
+  unknownOption: TripleBooleanRadioOption;
+};
+
+// Future podcast title??
```
Comment (jlhogan) [inline client/src/components/TripleBooleanRadio.tsx:19]: :rofl:

File: client/src/components/Forms/ChildInfo/Fields/DisabilityServices.tsx (hunk 1 above)
Comment (jlhogan) [reply client/src/components/Forms/ChildInfo/Fields/DisabilityServices.tsx:7]: don't know why/howi couldn't find it when i first reviewed this PR...... please excuse my brain!

**Review by jlhogan — APPROVED** [review]

**Review by JordanGuinn — APPROVED** [review]
Looks reasonable to me

File: client/src/components/Forms/Enrollment/Fields/Site.tsx [hunk 3]
```diff
@@ -24,27 +17,25 @@ export const SiteField = <T extends Enrollment | ChangeEnrollment>({
...
+          parseInt(e.target.value) || null,
         value: `${site.id}`,
+        id: `site-${site.id}`,
       }))}
-      status={(data, _, props) =>
-        getValidationStatusForField(
-          enrollmentAccessor(data),
-          enrollmentAccessor(data).at('site').path,
-          props
+      status={(_, dataDriller) =>
```
Comment (kmagida110) [inline client/src/components/Forms/Enrollment/Fields/Site.tsx:34]: How is this related to the radio button or is it not?

**Review by kmagida110 — APPROVED** [review]
One quick question but good to go

File: client/src/components/Forms/Enrollment/Fields/Site.tsx (hunk 3 above)
Comment (mmazanec22) [reply client/src/components/Forms/Enrollment/Fields/Site.tsx:34]: The radio button group was being treated as a field, which makes sense data-wise but is kinda messed up html-wise and props-passing-wise bc radio groups are actually a fieldset under the hood.  This PR (and the component library one a few days ago) makes it so that this is a fieldset instead, and therefore needed to take the fieldset status function.  I added the data driller as something passed to the status func via the component library fieldset status func and made a note that it should align better with the field status func in the future.  The data driller is used here to get the appropriate parent object and pass it to the status func.

File: client/src/components/TripleBooleanRadio.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  label: string;
+};
+
+export type TripleBooleanRadio<T> = {
+  field: keyof T;
+  id: string;
+  legend: string;
+  trueOption: TripleBooleanRadioOption;
+  falseOption: TripleBooleanRadioOption;
+  unknownOption: TripleBooleanRadioOption;
```
Comment (bamader) [inline client/src/components/TripleBooleanRadio.tsx:16]: I love this so much ❤️

**Review by bamader — APPROVED** [review]
Great stuff! Love that triple boolean refactor.
